x/observer/keeper/msg_server_vote_blame.go

```diff
@@ -13,52 +12,44 @@ import (
 
 func (k msgServer) VoteBlame(
 	goCtx context.Context,
-	vote *types.MsgVoteBlame,
+	msg *types.MsgVoteBlame,
 ) (*types.MsgVoteBlameResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	observationType := types.ObservationType_TSSKeySign
 
 	// GetChainFromChainID makes sure we are getting only supported chains , if a chain support has been turned on using gov proposal, this function returns nil
-	observationChain, found := k.GetSupportedChainFromChainID(ctx, vote.ChainId)
+	observationChain, found := k.GetSupportedChainFromChainID(ctx, msg.ChainId)
 	if !found {
-		return nil, cosmoserrors.Wrap(
+		return nil, sdkerrors.Wrapf(
 			crosschainTypes.ErrUnsupportedChain,
-			fmt.Sprintf("ChainID %d, Blame vote", vote.ChainId),
-		)
+			"ChainID %d, Blame vote", msg.ChainId)
 	}
 
-	if ok := k.IsNonTombstonedObserver(ctx, vote.Creator); !ok {
+	if ok := k.IsNonTombstonedObserver(ctx, msg.Creator); !ok {
 		return nil, types.ErrNotObserver
 	}
 
-	index := vote.Digest()
-	// Add votes and Set Ballot
-	// GetBallot checks against the supported chains list before querying for Ballot
-	ballot, isNew, err := k.FindBallot(ctx, index, observationChain, observationType)
+	ballot, isFinalized, isNew, err := k.VoteOnBallot(
+		ctx,
+		observationChain,
+		msg.Digest(),
+		types.ObservationType_TSSKeySign,
+		msg.Creator,
+		types.VoteType_SuccessObservation,
+	)
 	if err != nil {
-		return nil, err
+		return nil, sdkerrors.Wrap(err, errVoteOnBallot)
```

Comment:
Instead of using the const for error message, we can maybe precise which type of vote here?
```
failed to vote for blame
```

Reply:
Makes sense to me to specify the vote type when erroring out, applying it.